src/lib.rs

```diff
@@ -389,11 +389,13 @@ pub use crate::de::{from_slice, from_str, Deserializer, StreamDeserializer};
 #[doc(inline)]
 pub use crate::error::{Error, Result};
 #[doc(inline)]
-pub use crate::ser::{to_string, to_string_pretty, to_vec, to_vec_pretty};
+pub use crate::ser::{
+    to_string, to_string_pretty, to_vec, to_vec_pretty, to_writer, to_writer_pretty,
```

Comment:
https://github.com/serde-rs/json/pull/1122 describes some considerations about exposing no-std to_writer. What's done in this PR is not going to work.